src/api/lookups/calendars/calendars.js

```diff
@@ -14,70 +20,62 @@ const BASE_URL = '/calendars';
 const fieldsToSend = ['domainId', 'name', 'description', 'timezone', 'startAt', 'endAt', 'day',
     'accepts', 'excepts', 'startTimeOfDay', 'endTimeOfDay', 'disabled', 'date', 'repeat'];
 
-export const getCalendarList = async (page = 0, size = 10, search) => {
-    const domainId = store.state.userinfo.domainId || undefined;
-    if (search && search.slice(-1) !== '*') search += '*';
+const listGetter = new WebitelSDKListGetter(calendarService.searchCalendar);
+const itemGetter = new WebitelSDKItemGetter(calendarService.readCalendar);
+const timezoneGetter = new WebitelSDKListGetter(calendarService.searchTimezones);
+const itemCreator = new WebitelSDKItemCreator(calendarService.createCalendar, fieldsToSend);
+const itemUpdater = new WebitelSDKItemUpdater(calendarService.updateCalendar, fieldsToSend);
+const itemDeleter = new WebitelSDKItemDeleter(calendarService.deleteCalendar);
+
+itemGetter.responseHandler = (response) => {
     const defaultObject = {
-        _isSelected: false,
+        name: '',
+        timezone: {},
+        description: '',
+        startAt: Date.now(),
+        endAt: Date.now(),
+        expires: !!(response.startAt || response.endAt),
+        accepts: [],
+        excepts: [],
+        _dirty: false,
     };
-
-    try {
-        const response = await calendarService.searchCalendar(page, size, search, domainId);
-        if (response.items) {
-            return response.items.map(item => {
-                return {...defaultObject, ...item};
-            });
+    response.accepts = response.accepts.map(accept => {
+        return {
+            day: accept.day || 0,
+            disabled: accept.disabled || false,
+            start: accept.startTimeOfDay || 0,
+            end: accept.endTimeOfDay || 0
         }
-        return [];
-    } catch (err) {
-        throw err;
-    }
+    });
+    return {...defaultObject, ...response};
 };
 
-export const getCalendar = async (id) => {
-    const domainId = store.state.userinfo.domainId || undefined;
-
-    try {
-        let response = await calendarService.readCalendar(id, domainId);
-        const defaultObject = {
-            name: '',
-            timezone: {},
-            description: '',
-            startAt: Date.now(),
-            endAt: Date.now(),
-            expires: !!(response.startAt || response.endAt),
-            accepts: [],
-            excepts: [],
-            _dirty: false,
+timezoneGetter.responseHandler = (response) => {
```

Comment:
no need in timezone next because timezone is used only on dropdown's

Reply:
done